docs/gallery/advanced_io/streaming.py

```diff
@@ -1,98 +1,119 @@
-#
-# .. code-block:: python
-#
-#   from pynwb import NWBHDF5IO
-#
-#   with NWBHDF5IO(s3_path, mode='r', load_namespaces=True, driver='ros3') as io:
-#       nwbfile = io.read()
-#       print(nwbfile)
-#       print(nwbfile.acquisition['lick_times'].time_series['lick_left_times'].data[:])
-
-####################
-# Method 2: s3fs
-# ~~~~~~~~~~~~~~
-# s3fs is a library that creates a virtual filesystem for an S3 store. With this approach, a virtual file is created
-# for the file and virtual filesystem layer will take care of requesting data from the s3 bucket whenever data is
-# read from the virtual file.
-#
-# First install s3fs:
-#
-# .. code-block:: bash
-#
-#   pip install s3fs
-#
-# Then in Python:
-#
-# .. code-block:: python
-#
-#   import s3fs
-#   import pynwb
-#   import h5py
-#
-#   fs = s3fs.S3FileSystem(anon=True)
-#
-#   f = fs.open("s3://dandiarchive/blobs/43b/f3a/43bf3a81-4a0b-433f-b471-1f10303f9d35", 'rb')
-#   file = h5py.File(f)
-#   io = pynwb.NWBHDF5IO(file=file, load_namespaces=True)
-#
-#   io.read()
-#
-# The above snippet opens an arbitrary file on DANDI. You can use the ``DandiAPIClient`` to find the s3 path,
-# but you will need to adjust this url to give it a prefix of "s3://dandiarchive/" as shown above.
-#
-# The s3fs approach has the advantage of being more robust that ROS3. Sometimes s3 requests are interrupted,
-# and s3fs has internal mechanisms to retry these requests automatically, whereas ROS3 does not. However, it may not
-# be available on all platforms. s3fs does not currently work for Windows.
+   with NWBHDF5IO(s3_url, mode='r', load_namespaces=True, driver='ros3') as io:
+       nwbfile = io.read()
+       print(nwbfile)
+       print(nwbfile.acquisition['lick_times'].time_series['lick_left_times'].data[:])
 
+Streaming Method 2: fsspec
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+fsspec is another data streaming approach that is quite flexible and has several performance advantages. This library
+creates a virtual filesystem for remote stores. With this approach, a virtual file is created for the file and
+the virtual filesystem layer takes care of requesting data from the s3 bucket whenever data is
+read from the virtual file.
+
+First install ``fsspec`` and the dependencies of the :py:class:`~fsspec.implementations.http.HTTPFileSystem`:
+
+.. code-block:: bash
+
+   pip install fsspec requests aiohttp
+
+Then in Python:
+
+.. code-block:: python
+
+    import fsspec
+    import pynwb
+    import h5py
+    from fsspec.implementations.cached import CachingFileSystem
+
+    # first, create a virtual filesystem based on the http protocol and use
+    # caching to save accessed data to RAM.
+    fs = CachingFileSystem(
+        fs=fsspec.filesystem("http"),
+        # target_protocol='blockcache',
+        cache_storage="nwb-cache",
+        # cache_check=600,
+        # block_size=1024,
+        # check_files=True,
+        # expiry_times=True,
+        # same_names=True,
```

Comment:
I copy pasted those commented out options from older code in datalad-fuse. Not even sure if all of them applicable ATM, so best to just strip them away for now,

```suggestion
        cache_storage="nwb-cache",  # Local folder for the cache
```